storagemarket/deal_execution.go

```diff
@@ -186,6 +190,21 @@ func (p *Provider) execDealUptoAddPiece(ctx context.Context, pub event.Emitter,
 		p.dealLogger.Infow(deal.DealUuid, "deal has already been handed over to the sealing subsystem")
 	}
 
+	// Index deal in DAGStore and Announce deal
+	if deal.Checkpoint < dealcheckpoints.IndexedAndAnnounced {
+		if err := p.indexAndAnnounce(ctx, pub, deal); err != nil {
+			// any error here is always a recoverable error as this step is completely idempotent
```

Comment:
It looks like there's the possibility for a non-recoverable error also - for example adding a deal to the piece store (eg if there's a problem connecting to the datastore)

Reply:
It should never fail unless something has temporarily gone wrong.